File: sms/sms-user/src/main/sacla/com/yihu/cwd/user/dao/UserDao.scala
package com.yihu.cwd.user.dao

import com.yihu.cwd.user.model.User
import org.springframework.data.repository.PagingAndSortingRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor
/**
  * Created by Administrator on 2016.10.26.
  */
trait UserDao extends PagingAndSortingRepository[User,Integer] with JpaSpecificationExecutor[User]{

}

File: auth2/auth2-demo/src/main/java/com/cwd/auth2/dao/UserDao.java
package com.cwd.auth2.dao;

import com.cwd.auth2.model.User;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.PagingAndSortingRepository;

/**
 * Created by Administrator on 2017/1/11.
 */
public interface UserDao extends PagingAndSortingRepository<User, String>, JpaSpecificationExecutor<User> {

    User findByEmail(String lowercaseLogin);

    @Query("SELECT u FROM User u WHERE LOWER(u.username) = LOWER(:username)")
    User findByUsernameCaseInsensitive(String lowercaseLogin);
}
